### PR TITLE
feat: implement `From` array trait for `BoundedVec`

### PR DESCRIPTION
**Pull request by LogvinovLeon**
# Description

## Problem\*

Implementing a concise, single-line method for converting arrays to BoundedVec would enhance usability.

## Summary\*

This pull request implements the `From` trait to enable array-to-BoundedVec conversions, allowing for the following usage
```rust
let vec = BoundedVec::from([1, 2, 3]);
```

instead of:
```rust
let vec: BoundedVec<Field, 3> = BoundedVec::new();
vec.extend_from_array([1, 2, 3]);
```

## Documentation\*

Check one:
- [x] No documentation needed.
- [ ] Documentation included in this PR.
- [ ] **[For Experimental Features]** Documentation to be submitted in a separate PR.

# PR Checklist\*

- [x] I have tested the changes locally.
- [x] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` on default settings.

### CODE REVIEW

File: noir_stdlib/src/collections/bounded_vec.nr [hunk 1]
```diff
@@ -128,4 +137,45 @@ mod bounded_vec_tests {
...
+
+        #[test]
+        fn equal_len() {
+            let array = [1, 2, 3];
+            let bounded_vec = BoundedVec::from(array);
+
+            assert_eq(bounded_vec.max_len(), 3);
+            assert_eq(bounded_vec.len(), 3);
+            assert_eq(bounded_vec.storage(), array);
+        }
```
Comment (jfecher) [inline noir_stdlib/src/collections/bounded_vec.nr:162]: I can't seem to figure out how MaxLen is inferred to be 3 here despite `MaxLen` and `ArrayLen` being separate in the From impl. Yet I've tested this myself and it passes.

**Review by jfecher — COMMENTED** [review]
Looks good, two things:

1. Can you create a `from_array` function as well for users that prefer a more explicit constructor? This helps users find this function in the docs as well.
2. Do you mind adding these to the docs? They are in `docs/docs/noir/standard_library/containers/boundedvec.md`. You should be able to copy & edit an existing entry for the formatting.

**Comment by TomAFrench** [comment]
Merging master will fix the macos errors.

**Comment by akonior** [comment]
@jfecher I've added `from_array` method directly to `BoundedVec` `impl` and updated documentation.

**Review by jfecher — APPROVED** [review]
Thank you!

**Pull request merged**